src/android/MediaCapture.java

```diff
@@ -204,48 +71,55 @@ public void run() {
         }
     }
 
-    @Override
-    public void onPause(boolean multitasking) {
-        if (previewing) {
-            this.appPausedWithActivePreview = true;
-            this.pausePreview(null);
+    private void openNewActivity(Context context) {
+
+        if (!hasPermission()) {
+            requestPermission(33);
+            return;
+        } else {
+            Intent intent = new Intent(context, MainActivity.class);
+            intent.putExtra("VIDEO_MAX_LENGTH", lengthInSeconds);
+            intent.putExtra("NEXT_VIDEO_URL", createCaptureFile().getAbsolutePath());
+            cordova.startActivityForResult((CordovaPlugin) this ,intent,VIDEO_URL);
         }
     }
 
-    @Override
-    public void onResume(boolean multitasking) {
-        if (this.appPausedWithActivePreview) {
-            this.appPausedWithActivePreview = false;
-            this.resumePreview(null);
-        }
+    private File createCaptureFile() {
+        String timeStamp = new SimpleDateFormat("yyyyMMdd_HHmmss").format(new Date());
+        return new File(getTempDirectoryPath(),
+                "vid_" + timeStamp + ".mp4");
     }
 
-    private boolean hasFlash() {
-        if (flashAvailable == null) {
-            flashAvailable = false;
-            final PackageManager packageManager = this.cordova.getActivity().getPackageManager();
-            for (final FeatureInfo feature : packageManager.getSystemAvailableFeatures()) {
-                if (PackageManager.FEATURE_CAMERA_FLASH.equalsIgnoreCase(feature.name)) {
-                    flashAvailable = true;
-                    break;
-                }
-            }
+    private String getTempDirectoryPath() {
+        File cache = null;
+
+        // SD Card Mounted
+        if (Environment.getExternalStorageState().equals(Environment.MEDIA_MOUNTED)) {
```

Comment:
SD card situation does it actually work when reading the file and deleting the file from storage?

Reply:
to be fixed later if needed